core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -2852,16 +2881,34 @@ wasi_ssp_sock_addr_local(
 #if !defined(WASMTIME_SSP_STATIC_CURFDS)
     struct fd_table *curfds,
 #endif
-    __wasi_fd_t fd, uint8 *buf, __wasi_size_t buf_len)
+    __wasi_fd_t fd, __wasi_addr_t *addr)
 {
     struct fd_object *fo;
+    uint8_t buf[16];
+    __wasi_ip_port_t port;
+    uint8_t is_ipv4;
```

Comment:
Had better use `uint8`, we usually use type uint32/uint16/int8 instead of uint32_t/uint16_t/uint8_t inside runtime.